packages/components/src/pagination-control/README.md

```diff
@@ -0,0 +1,29 @@
+# pagination-control
+
+This component returns a `ul` of dots used to navigate between slides, pages, or steps. The color is inherited from the WP-admin theme color.
+
+## Usage
+
+```jsx
+import { PaginationControl } from '@automattic/components';
+
+function MyComponent() {
+	return (
+		<PaginationControl
+			currentPage={ currentPage }
+			numberOfPages={ numberOfPages }
+			setCurrentPage={ setCurrentPage }
```

Comment:
Does `current` [add any meaning here](p7H4VZ-2LV-p2#:~:text=in%20OOP%20languages.-,Interestingly,-%2C%20since%20these%20words)?

Since `setCurrentPage` expects some sort of handler, usually it's prefixed with `on`. `onChange` would be a great name IMHO.

Reply:
This is the handler is meant to be the `onClick` event so maybe it would be better to name it as such? I also updated the README to note that the `index` is being passed in as an argument.

Reply:
Hmm I would expect a callback that reflects an `onClick`/event handler to also pass along the `event` though?

Reply:
Yex exactly. onClick should forward the event, onChange should give the new page.

Reply:
As of currentPage. I 100% think `pageIndex` is much better, because it conveys the numeric nature of the value, and gets rid of `current` which I'm [heavily biased against](p7H4VZ-2LV-p2#:~:text=Interestingly%2C%20since%20these,is%20probably%20sufficient.).

Reply:
> which I'm biased against (see p7H4VZ-2LV-p2)

Nice. Not sure I agree though personally. I find that to be a little heavy-handed as a general principle. I think context and granularity in naming are important. It also precludes confusion when the same reference needs to exist in a different context or in a more general/finer way. Here, "current" communicates a clear semantic ("this is the current/active index"), whilst "index" is more ambiguous (needs some form of disambiguation, which could result in a docstring "the current index" or so).

Not sure there's anything wrong with the semantics here:

```
currentIndex = 3;
for index in indices:
    if index  == currentIndex, do something;
```

But, for the sake of argument. Naming is a hard problem.

Reply:
>Yes exactly. I think onClick should forward the event, onChange should give the new page.

Just so I understand, we are only talking about changing the name of the prop from `setCurrentPage` to `onChange`? This is separate from actually changing the actual button handle from `onClick` to `onChange`?

Reply:
> I agree more with this train of thought but I do think current may not be the best descriptor. How about going with activePageIndex? This will define what "index" while offering a more specific descriptor to avoid confusion

Sorry, just saw the response. `activePageIndex` sounds descriptive/accurate enough to me too if you feel like going with that :-)